keps/sig-node/4176-cpumanager-spread-cpus-preferred-policy/README.md

```diff
@@ -0,0 +1,824 @@
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+`full-pcpus-only` was introduced to resolve the problem that different containers sharing the same physical cores which leads noisy neighbor issue. `distribute-cpus-across-numa` distributed cpus across NUMA to make sure no single worker suffers from NUMA effects more than any other, improving the overall performance of these types of applications. These two and default behavior can not meet the requirement of some applications which need to take advantage of L2 Cache. In that case, we need to spread hyper threads across physical cores, while NUMA is not a important factor. In such cases, the assumption is that the single physical core with two applications won't be always busy. So we can take advantage of CPU Cache to improve the performance of these applications. Otherwise, it still have the noisy neighbor issue.
+
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Introduce a new CPU Manager Static Policy that spreads hyper threads across physical cores without considering NUMA.
+- Enhance application performance by taking advantage of L2 Cache.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- This proposal does not aim to modify the existing CPU Manager Core Binding Policies. It focuses solely on introducing a new policy for spreading hyper threads across physical cores.
+- It does not address other resource allocation or management aspects within Kubernetes.
+
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+We propose to add a new `CPUManager` policy option called `spread-physical-cpus-preferred` to the static CPUManager policy. When enabled, this will trigger the CPUManager to try to allocate CPUs across physical nodes as much as possible.
```

Comment:
```suggestion
We propose to add a new `CPUManager` policy option called `distribute-cpus-across-cores` to the static CPUManager policy. When enabled, this will trigger the CPUManager to try to allocate CPUs across physical cores as much as possible. It will not prohibit a CPU from being allocated on a core that already has a CPU allocated, but it will only resort to doing so once there is no other option.
```